diff -U 3 -H -d -r -N -- openldap-2.4.40/contrib/ldapc++/configure openldap-2.4.40_patch/contrib/ldapc++/configure
--- openldap-2.4.40/contrib/ldapc++/configure	2014-09-19 05:48:49.000000000 +0400
+++ openldap-2.4.40_patch/contrib/ldapc++/configure	2014-12-12 19:39:49.832222000 +0300
@@ -19151,57 +19151,6 @@
 
 fi
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for ldap_add_ext in -lldap" >&5
-$as_echo_n "checking for ldap_add_ext in -lldap... " >&6; }
-if test "${ac_cv_lib_ldap_ldap_add_ext+set}" = set; then :
-  $as_echo_n "(cached) " >&6
-else
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lldap
-    -llber
-     $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char ldap_add_ext ();
-int
-main ()
-{
-return ldap_add_ext ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_ldap_ldap_add_ext=yes
-else
-  ac_cv_lib_ldap_ldap_add_ext=no
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_ldap_ldap_add_ext" >&5
-$as_echo "$ac_cv_lib_ldap_ldap_add_ext" >&6; }
-if test "x$ac_cv_lib_ldap_ldap_add_ext" = x""yes; then :
-
-        :
-
-else
-
-        echo "        didn't find ldap_add_ext in libldap !";
-        echo "        Check for the right version (>= 2.0) of the OpenLDAP libraries";
-        echo "        or try the --with-libldap option.";
-        exit
-
-fi
-
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether time.h and sys/time.h may both be included" >&5
 $as_echo_n "checking whether time.h and sys/time.h may both be included... " >&6; }
 if test "${ac_cv_header_time+set}" = set; then :
